src/query_parser/parser.y

```diff
@@ -278,13 +306,30 @@ text_expr(A) ::= text_expr(B) text_expr(C) . [AND] {
     }
 }
 
+text_expr(A) ::= text_expr(B) text_expr(C) . [AND] {
```

Comment:
Can also add a comment to explain that this rule is required since it is specifically using only `text_expr` non-terminals,
So it could reduce them to the more specific `text_expr` and not to a general `expr`, so we could maintain the differentiation between `text_expr` and `expr` when possible.